diff --git a/docroot/modules/contrib/paragraphs/paragraphs.install b/docroot/modules/contrib/paragraphs/paragraphs.install
index 3accef3bb..f83d06ea1 100644
--- a/docroot/modules/contrib/paragraphs/paragraphs.install
+++ b/docroot/modules/contrib/paragraphs/paragraphs.install
@@ -338,19 +338,10 @@ function paragraphs_update_8016() {
 }
 
 /**
- * Update the revision metadata keys to remove revision uid.
- */
+* Moved function content to paragraphs_update_8023().
+* see https://www.drupal.org/project/paragraphs/issues/2833935#comment-13471948
+*/
 function paragraphs_update_8017() {
-  $entity_definition_update_manager = \Drupal::entityDefinitionUpdateManager();
-  $entity_type = $entity_definition_update_manager->getEntityType('paragraph');
-  if ($metadata_keys = $entity_type->get('revision_metadata_keys')) {
-    if (is_array($metadata_keys) && isset($metadata_keys['revision_user'])) {
-      unset($metadata_keys['revision_user']);
-      $metadata_keys['revision_default'] = 'revision_default';
-      $entity_type->set('revision_metadata_keys', $metadata_keys);
-      $entity_definition_update_manager->updateEntityType($entity_type);
-    }
-  }
 }
 
 /**
@@ -479,3 +470,19 @@ function paragraphs_update_8022() {
   // Regenerate entity type indexes.
   $manager->updateEntityType($manager->getEntityType('paragraph'));
 }
+
+/**
+ * Update the revision metadata keys to remove revision uid.
+ */
+function paragraphs_update_8023() {
+  $entity_definition_update_manager = \Drupal::entityDefinitionUpdateManager();
+  $entity_type = $entity_definition_update_manager->getEntityType('paragraph');
+  if ($metadata_keys = $entity_type->get('revision_metadata_keys')) {
+    if (is_array($metadata_keys) && isset($metadata_keys['revision_user'])) {
+      unset($metadata_keys['revision_user']);
+      $metadata_keys['revision_default'] = 'revision_default';
+      $entity_type->set('revision_metadata_keys', $metadata_keys);
+      $entity_definition_update_manager->updateEntityType($entity_type);
+    }
+  }
+}
