diff --git a/docroot/modules/contrib/button_link/src/Plugin/Field/FieldFormatter/ButtonLinkFormatter.php b/docroot/modules/contrib/button_link/src/Plugin/Field/FieldFormatter/ButtonLinkFormatter.php
index 99335d78..778c5648 100644
--- a/docroot/modules/contrib/button_link/src/Plugin/Field/FieldFormatter/ButtonLinkFormatter.php
+++ b/docroot/modules/contrib/button_link/src/Plugin/Field/FieldFormatter/ButtonLinkFormatter.php
@@ -58,6 +58,12 @@ public function settingsForm(array $parentForm, FormStateInterface $form_state)
         'btn-info' => $this->t('Info'),
         'btn-warning' => $this->t('Warning'),
         'btn-danger' => $this->t('Danger'),
+        'btn-outline-default' => $this->t('Outline Default'),
+        'btn-outline-primary' => $this->t('Outline Primary'),
+        'btn-outline-success' => $this->t('Outline Success'),
+        'btn-outline-info' => $this->t('Outline Info'),
+        'btn-outline-warning' => $this->t('Outline Warning'),
+        'btn-outline-danger' => $this->t('Outline Danger'),
       ],
       '#required' => TRUE,
     ];
