diff --git a/docroot/.htaccess b/docroot/.htaccess
index 9a73a3d3a..e7c53ee78 100644
--- a/docroot/.htaccess
+++ b/docroot/.htaccess
@@ -76,6 +76,23 @@ AddEncoding gzip svgz
   RewriteCond %{HTTPS} on
   RewriteRule ^ - [E=protossl:s]
 
+  # Redirect to bare domain to www.prefix domain using ssl (except for Acquia
+  # alternative sites like test or dev and Lando dev sites.)
+  RewriteCond %{HTTP_HOST} !\.acquia-sites\.com [NC]
+  RewriteCond %{HTTP_HOST} !\.lndo\.site [NC]
+  RewriteCond %{HTTP_HOST} !=dev.maseniorcare.org [NC]
+  RewriteCond %{HTTP_HOST} !=stage.maseniorcare.org [NC]
+  RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
+
+  # Rewrite non-ssl traffic to https:// prefix  (except for acquia
+  # alternative sites like test or dev.)
+  RewriteCond %{HTTP_HOST} !\.acquia-sites\.com [NC]
+  RewriteCond %{HTTP_HOST} !\.lndo\.site [NC]
+  RewriteCond %{HTTP_HOST} !=dev.maseniorcare.org [NC]
+  RewriteCond %{HTTP_HOST} !=stage.maseniorcare.org [NC]
+  RewriteCond %{HTTP:X-Forwarded-Proto} !https
+  RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]
+
   # Make sure Authorization HTTP header is available to PHP
   # even when running as CGI or FastCGI.
   RewriteRule ^ - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
